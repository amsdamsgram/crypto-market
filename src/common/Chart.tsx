import React, { Component } from "react";
import { StyleSheet, View } from "react-native";
import { Grid, LineChart, YAxis } from "react-native-svg-charts";

import Txt from "../common/Txt";
import Theme from "../Theme";

interface IProps {
  color: string;
  data: number[];
  height: number;
  title: string;
}

const AXE_FONT_SIZE = 10;

const styles = StyleSheet.create({
  chart: {
    flex: 1,
    marginLeft: 10
  },
  container: {
    flexDirection: "row",
    padding: Theme.globalStyles.padding
  },
  header: {
    alignItems: "center",
    justifyContent: "center",
    padding: Theme.globalStyles.padding
  },
  title: {
    fontSize: 20,
    fontWeight: "500"
  }
});

export default class Chart extends Component<IProps> {
  renderHeader() {
    return (
      <View style={styles.header}>
        <Txt style={styles.title}>{this.props.title}</Txt>
      </View>
    );
  }
  render() {
    const verticalContentInset = { top: 10, bottom: 10 };

    return (
      <View>
        {this.renderHeader()}
        <View style={[styles.container, { height: this.props.height }]}>
          <YAxis
            data={this.props.data}
            contentInset={verticalContentInset}
            svg={{
              fill: Theme.colors.textSecondary,
              fontSize: AXE_FONT_SIZE
            }}
          />
          <LineChart
            style={styles.chart}
            data={this.props.data}
            contentInset={verticalContentInset}
            svg={{ stroke: Theme.colors.primary }}
          >
            <Grid />
          </LineChart>
        </View>
      </View>
    );
  }
}
