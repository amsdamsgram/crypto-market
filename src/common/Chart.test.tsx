import React from "react";
import renderer from "react-test-renderer";

import Chart from "./Chart";

describe("Chart", () => {
  test("should match snapshot", () => {
    const tree = renderer
      .create(
        <Chart
          color="white"
          data={[1, 2, 3, 4, 5]}
          height={100}
          title="Title"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
