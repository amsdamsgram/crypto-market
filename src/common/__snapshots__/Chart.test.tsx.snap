// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart should match snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#414141",
            "fontSize": 14,
          },
          Object {
            "fontSize": 20,
            "fontWeight": "500",
          },
        ]
      }
    >
      Title
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "padding": 15,
        },
        Object {
          "height": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          undefined,
        ]
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flexGrow": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "transparent",
              "fontSize": 10,
            }
          }
        >
          4.5
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "marginLeft": 10,
        }
      }
    >
      <View
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
